transports/upgrades/sqlserver-4to5.md

```diff
@@ -0,0 +1,118 @@
+var delayedDelivery = transport.UseNativeDelayedDelivery();
+
+delayedDelivery.BatchSize(100);
+delayedDelivery.DisableTimeoutManagerCompatibility()
+delayedDelivery.ProcessingInterval(TimeSpan.FromSeconds(5));
+delayedDelivery.TableSuffix("Delayed");
+```
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+var delayedDelivery = transport.NativeDelayedDelivery();
+
+delayedDelivery.BatchSize(100);
+delayedDelivery.DisableTimeoutManagerCompatibility()
+delayedDelivery.ProcessingInterval(TimeSpan.FromSeconds(5));
+delayedDelivery.TableSuffix("Delayed");
+```
+
+
+## Native publish subscribe
+
+SQL Server transport version 5 introduces native support for the publish subscribe pattern. It does so via a dedicated subscription routing table which holds subscription information for each event type. When an endpoint subscribes to an event, an entry is created in the subscription routing table. When an endpoint publishes an event, the subscription routing table is queried to find all of the subscribing endpoints.
+
+Upgrade a single endpoint at a time. As each endpoint is upgraded to version 5, configure it to operate in backwards compatibility mode to enable it to participate in publish subscribe operations with endpoints that are running on version 4.x or below. An endpoint operating in backwards compatibility mode will check the configured persistence and the native subscriptions table for subscribers when publishing a message. The transport will only send a single copy of the message to each subscriber.
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+
+var pubSubCompat = transport.EnableMessageDrivenPubSubCompatibilityMode();
+```
+
+Configuration APIs to register publishers for event types has been moved to the compatibility mode settings object.
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+var routing = transport.Routing();
+routing.RegisterPublisher(eventType, publisherEndpoint);
+routing.RegisterPublisher(assembly, publisherEndpoint);
+routing.RegisterPublisher(assembly, namespace, publisherEndpoint);
+routing.SubscriptionAuthorizer(authorizationFunction);
+```
+
+```csharp
+var transport = endpointConfiguration.UseTransport<SqlServerTransport>();
+
+var pubSubCompat = transport.EnableMessageDrivenPubSubCompatibilityMode();
+pubSubCompat.RegisterPublisher(eventType, publisherEndpoint);
+pubSubCompat.RegisterPublisher(assembly, publisherEndpoint);
+pubSubCompat.RegisterPublisher(assembly, namespace, publisherEndpoint);
+pubSubCompat.SubscriptionAuthorizer(authorizationFunction);
+```
+
+NOTE: The `routing.DisablePublishing()` API has been deprecated and should be removed. This API was created to allow an endpoint to run without a configured subscription persistence. On version 5 and above, a subscription persistence is not required unless the endpoint runs in backwards compatibility mode.
```

Comment:
@SzymonPobiega are there scenarios where DisablePublishing still makes sense when operating in compatibility mode? i.e. We had this before so we still don't need to configure a subscription persistence but we want to subscribe to events coming from other endpoints.

Reply:
`SubscriptionMigrationMode` in the core does not look at the `DisablePublishing` settings so without changing the core we would not be able to make use of that feature -- in the compatibility mode subscription persistence is mandatory. I think that what you wrote is OK 👍